src/JsonRpcEngine.ts

```diff
@@ -57,6 +60,26 @@ export class JsonRpcEngine extends SafeEventEmitter {
     this._middleware.push(middleware as JsonRpcMiddleware<unknown, unknown>);
   }
 
+  /**
+   * Calls the `destroy()` function of any middleware with that property and clears
+   * the middleware array.
+   */
+  cleanup(): void {
```

Comment:
[nit] Once this method is called, can the instance of JsonRpcEngine continue to be used? If not, should we call this destroy too?